### ISSUE TITLE
rigs disconnecting - 2 issues: mining.extranonce.subscribe and mining.multi_version

### ISSUE THREAD

**Issue by xborow** [bonesoul/CoiniumServ#1042]
rigs still disconnecting 2 reasons:

05/27/2018 23:31:29 +02:00 [Debug] [StratumServer] [Bitcoin] Stratum client connected: "188.166.37.58:17921"
05/27/2018 23:31:29 +02:00 [Debug] [StratumServer] [Bitcoin] Stratum client connected: "188.166.37.58:17923"
05/27/2018 23:31:31 +02:00 [Debug] [MinerManager] [Bitcoin] Authenticated miner: zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz[188.166.37.58:17921]
05/27/2018 23:31:31 +02:00 [Debug] [MinerManager] [Bitcoin] Authenticated miner: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx[188.166.37.58:17923]
05/27/2018 23:31:31 +02:00 [Debug] [StratumServer] [Bitcoin] Stratum client disconnected: "Not Connected!"
05/27/2018 23:31:31 +02:00 [Debug] [StratumServer] [Bitcoin] Stratum client disconnected: "Not Connected!"

1st issue bfminer:
-------------------
}"
05/27/2018 23:31:31 +02:00 [Verbose] [StratumMiner] [Bitcoin] rx: "{
    'id' : 91425,
    'method' : 'mining.extranonce.subscribe',
    'params' : [
        
    ]
}"
05/27/2018 23:31:31 +02:00 [Verbose] [StratumMiner] [Bitcoin] tx: "{
    'jsonrpc' : '2.0',
    'error' : {
        'code' : -32601,
        'message' : 'Method not found',
        'data' : 'The method does not exist / is not available.'
    },
    'id' : 91425
}"

2nd issue Antminer s9:
-------------------------
05/27/2018 23:31:31 +02:00 [Verbose] [StratumMiner] [Bitcoin] rx: "{
    'id' : ,
    'method' : 'mining.multi_version',
    'params' : [

    'jsonrpc' : '2.0',
    'error' : {
        'code' : ,
        'message' : 'Method not found',
        'data' : 'The method does not exist / is not available.'

article related to 2nd issue ASICBOOST:
https://www.reddit.com/r/Bitcoin/comments/63yo27/some_circumstantial_evidence_supporting_the_claim/dfy5o65/

would be possible to implement kind of fix for the above 2 issues, please?
or anybody how can advice which lines should be modified to avoid the above two issues, please?

**Comment by robbterr** [comment]
The fix should be something like this:
https://github.com/robbterr/CoiniumServ/commit/7b023078f390cdee70aaaeb9f9f9bf712109e896

I did´t want to do a pull request, because i have no idea what i am doing, its the first time i did something with C# :)

**Comment by leandrogomesmachado** [comment]
https://github.com/robbterr/CoiniumServ/commit/7b023078f390cdee70aaaeb9f9f9bf712109e896

Works very well

**Comment by xborow** [comment]
Sounds like a solution,  have tested, but still getting miner connected, disconnected, connected, disconnected, no error, tested xnonce a and other rigs from miningrigrentals.com, so migrared to miningcore instead, works much better

A bit sad beacause Coiniumserv looks nice to me, even webfronted in place, but if it does not work with miners for bitcoin, first and basic coin then its useless

And i'm really sure it's not beacuse of miners/rigs, with miningcore, all rigs rented from miningrigrentals.com connect without no issues

**Comment by leandrogomesmachado** [comment]
I don't remember what I have done but I solved the problem with connecting/disconnecting.

Now I have a problem with a hybrid BITCOIN/ETHEREUM coin that doesn't work to submit blocks!

But this solution have worked for me (the link that I have put above)

**Comment by xborow** [comment]
Ahaaa, you right, it would work if i read focused, the code make sense and it should fix the issue, but there are misstypes, "MiningExtraonceSubscrube" it should be "MiningExtranonceSubscribe"

need to check first if i'm right, but hope there is no code error for submitting blocks ;)

**Comment by xborow** [comment]
Nope, even if added those lines and fixed misstypes there, still the same issue for miners, for sure something wrong with Coiniumserv:

23:56:28 [Debug] [StratumServer] [Bitcoin] Stratum client connected: "45.76.227.84:17704"
23:56:31 [Debug] [MinerManager] [Bitcoin] Authenticated miner: XYXYXYXY [45.76.227.84:17704]
23:56:31 [Debug] [StratumServer] [Bitcoin] Stratum client disconnected: "Not Connected!"
23:56:36 [Debug] [StratumServer] [Bitcoin] Stratum client connected: "45.76.227.84:17857"
23:56:39 [Debug] [MinerManager] [Bitcoin] Authenticated miner: XYXYXYXY [45.76.227.84:17857]
23:56:39 [Debug] [StratumServer] [Bitcoin] Stratum client disconnected: "Not Connected!"
23:56:45 [Debug] [StratumServer] [Bitcoin] Stratum client connected: "45.76.227.84:18084"
23:56:47 [Debug] [MinerManager] [Bitcoin] Authenticated miner: XYXYXYXY [45.76.227.84:18084]
23:56:47 [Debug] [StratumServer] [Bitcoin] Stratum client disconnected: "Not Connected!"
Btw address changed by XYXYXY ;)

As soon as changed to miningcore, the miner auth and sharing shares..
